core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1281,19 +1285,24 @@ protected Scope visitTable(Table table, Optional<Scope> scope)
                 // This is could be a reference to a logical view or a table
                 Optional<ConnectorViewDefinition> optionalView = metadata.getView(session, name);
                 if (optionalView.isPresent()) {
+                    analysis.addEmptyColumnReferencesForTable(accessControl, session.getIdentity(), name);
                     return createScopeForView(table, name, scope, optionalView.get());
                 }
-                tableHandle = metadata.getTableHandle(session, name);
+
+                // This can only be a table
+                tableHandle = metadata.getRedirectedTableHandle(session, name);
+                targetTableName = metadata.getRedirectedTableName(session, name);
+                analysis.addEmptyColumnReferencesForTable(accessControl, session.getIdentity(), targetTableName);
             }
 
             if (tableHandle.isEmpty()) {
-                if (metadata.getCatalogHandle(session, name.getCatalogName()).isEmpty()) {
```

Comment:
Let's leave this `if` is and add another and additional check (for redirected table name) if source catalog and schema exists

Reply:
* for tables that haven't been redirected,  the flow is untouched. `targetTableName` is same as `name` and exception handling happens here for catalog, schema, table existence (like before).

* for redirected tables, imo source schema existence shouldn't matter - as long as the connector can redirect to a target. `MetadataManager#getRedirectedTableHandle`  throws exception if (1) catalog doesn't exist during redirection at any step OR (2) getTableHandle fails on the target table due to catalog/schema/table existence issue, and reported accordingly.

If source catalog doesn't exist, the redirection will stop there and reported as "source catalog doesn't exist", which is the current behavior. 

even though there's some code-change, I think scope of the flow-change is small here, as you had suggested earlier. Redirection related logic/exception-handling is contained in the MetadataManager redirection apis. 

So IMO it'd be better to not duplicate the checks here for source and target both. Thoughts?